File: Week_02/590/n_ary_tree_postorder_traversal2.py
from typing import List


class Node:
    def __init__(self, val=None, children=None):
        self.val = val
        self.children = children


class Solution:
    def postorder(self, root: 'Node') -> List[int]:
        res = []

        stack = [root]
        while len(stack) > 0:
            node = stack.pop()
            if node is not None:
                res.append(node.val)
                for c in node.children:
                    stack.append(c)
        return res[::-1]

File: Week_02/590/n_ary_tree_postorder_traversal2.go
package main

func postorder2(root *Node) []int {
	res := make([]int, 0)
	stack := []*Node{root}

	for len(stack) > 0 {
		node := stack[len(stack)-1]
		stack = stack[:len(stack)-1]
		if node != nil {
			res = append(res, node.Val)
			for i := 0; i < len(node.Children); i++ {
				stack = append(stack, node.Children[i])
			}
		}
	}

	for i, j := 0, len(res)-1; i < j; i, j = i+1, j-1 {
		res[i], res[j] = res[j], res[i]
	}
	return res
}
